### DISCUSSION TITLE
convert objc.Object back to NSMenuItem

### DISCUSSION THREAD

**Discussion by nicboul** [progrium/darwinkit#29] [Q&A] [answered]
I'm programmatically adding items to the menu and I need to have more information inside the Method when the method is getting called. I added what I need in comment in the code here:

		item := cocoa.NSMenuItem_New()
		item.SetTitle("my super title")
		item.SetAction(objc.Sel("somethingUnique:"))
		cocoa.DefaultDelegateClass.AddMethod("somethingUnique:", func(myObject objc.Object) {
			// HERE, is it possible to convert "myObject" back to the NSMenuItem object ?
                         // I want to have access to the item Title
		})

**Comment by nicboul** [comment, marked as answer]
ok if I do "item.Item()" directly in the function it does what I want.... :)

**Reply by progrium** [reply]
yes, you can also wrap any `objc.Object` in any specific type wrapper like, for example, `NSMenuItem{myObject}`, but be sure its the right type. you should be able to at least check its class/type now with the `Class()` method on Object.
